examples/src/boids/boid.rs

```diff
@@ -53,7 +53,12 @@ impl Boid {
     }
 }
 
-pub fn game_tick(dt: f32, cursor: mint::Point2<f32>, boid: &mut Boid, other_boids: Vec<Boid>) {
+pub fn game_tick<'a>(
+    dt: f32,
+    cursor: mint::Point2<f32>,
+    boid: &'a mut Boid,
+    other_boids: (&'a mut [Boid], &'a mut [Boid]),
```

Comment:
I understand the advantage of passing a mutable reference to an array of boids rather than a vector in this case, but what is the advantage of partitioning into two arrays? There's a balance between code readability and performance, and I'd like to keep the '/examples' folder as easy-to-follow as possible. If not partitioning the code results in a significant performance disadvantage, evidenced by increased CPU and memory consumption, then I think it's at least worth mentioning in the comments and potentially including in the code. However, at this point, without further knowledge of the performance gains, it seems to me like an unnecessary or premature optimization.

Would you mind sharing your thoughts around this? :)

Reply:
Well the point of partitioning into slices is that it is the only way to express to the borrow checker that you will in fact only "touch" the memory before and after a certain position in the array. Otherwise you'd be passing multiple mutable references to the same structure, which is a big no-no according to the ownership rules.